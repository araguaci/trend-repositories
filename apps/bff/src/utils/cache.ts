const LRU = require('lru-cache');

export let cache = null;

export const initCache = () => {
  const options = {
    max: 500,
    ttl: 86400 * 1000, // 1 day
  };

  cache = new LRU(options);
};
