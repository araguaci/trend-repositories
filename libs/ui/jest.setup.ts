import '@testing-library/jest-dom';
jest.mock('next/dist/client/router', () => require('next-router-mock'));
