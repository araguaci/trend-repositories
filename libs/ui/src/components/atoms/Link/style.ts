import styled from 'styled-components';

export const StyledLink = styled.a`
  display: block;
  font-size: ${(props) => props.theme.sizes.large};
  color: ${(props) => props.theme.colors.blue_2};
  font-weight: ${(props) => props.theme.fontWeight.bold};
  cursor: pointer;

  &:hover {
    text-decoration: underline;
  }
`;
