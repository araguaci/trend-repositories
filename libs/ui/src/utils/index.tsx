/* istanbul ignore file */
import { ThemeProvider } from 'styled-components';
import theme from '../theme';
import { ReactNode } from 'react';
import { render, act } from '@testing-library/react';
import { MockedProvider } from '@apollo/client/testing';
import { MockedResponse } from '@apollo/client/testing/core';
import { InMemoryCache } from '@apollo/client/cache';

export const renderWithProvider = (node: ReactNode) => {
  return render(<ThemeProvider theme={theme}>{node}</ThemeProvider>);
};

export const renderWithMockProvider = (
  node: ReactNode,
  cache: InMemoryCache,
  mocks: ReadonlyArray<MockedResponse>
) => {
  return render(
    <ThemeProvider theme={theme}>
      <MockedProvider cache={cache} mocks={mocks}>
        {node}
      </MockedProvider>
    </ThemeProvider>
  );
};

export const renderWithThemeProvider = (node: ReactNode) => {
  return render(<ThemeProvider theme={theme}>{node}</ThemeProvider>);
};

export const waitForResponse = async () =>
  await act(async () => await new Promise((resolve) => setTimeout(resolve, 0)));
export * from '@testing-library/react';
