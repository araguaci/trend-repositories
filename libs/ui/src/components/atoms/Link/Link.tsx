import { FC } from 'react';
import { StyledLink } from './style';
import { default as NextLink, LinkProps } from 'next/link';

const Link: FC<
  Partial<LinkProps> & { className?: string; onClick?: () => void }
> = ({ children, href, onClick, className, ...props }) => {
  if (href) {
    return (
      <NextLink href={href} passHref>
        <StyledLink className={className} {...(props as never)}>
          {children}
        </StyledLink>
      </NextLink>
    );
  }

  return (
    <StyledLink onClick={onClick} className={className} {...(props as never)}>
      {children}
    </StyledLink>
  );
};

export default Link;
