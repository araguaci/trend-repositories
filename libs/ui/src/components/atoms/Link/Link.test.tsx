import Link from './Link';
import { renderWithProvider } from '@trend-repositories/ui/utils';
import { fireEvent, screen } from '@testing-library/react';

describe('Link Component', () => {
  it('should be rendered correctly', () => {
    const { asFragment } = renderWithProvider(
      <Link href="https://github.com">linguistic</Link>
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('should be work with callback', () => {
    // Arrange
    const handleClick = jest.fn();
    renderWithProvider(<Link onClick={handleClick}>linguistic</Link>);

    const link = screen.getByText(/linguistic/);

    //Act
    fireEvent.click(link);

    // Assertion
    expect(handleClick).toHaveBeenCalled();
  });
});
