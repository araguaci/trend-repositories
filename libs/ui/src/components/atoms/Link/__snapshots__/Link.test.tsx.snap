// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link Component should be rendered correctly 1`] = `
<DocumentFragment>
  <a
    class="style__StyledLink-ikww7b-0 fUma-dV"
    href="https://github.com"
  >
    linguistic
  </a>
</DocumentFragment>
`;
